edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java

```diff
@@ -151,6 +212,48 @@ public static List<Slot> getSchedule(Group group, Interval timerange) {
         }
         return schedule.filterEverySlotOutsideTimerange(allSlotOfGroup, timerange);
     }
+    /**
+     * Round to certain number of decimals
+     * Author:jav-rock
+     * @param d
+     * @param decimalPlace
+     * @return
+     */
+    public static float round(float d, int decimalPlace) {
+        BigDecimal bd = new BigDecimal(Float.toString(d));
+        bd = bd.setScale(decimalPlace, RoundingMode.FLOOR);
+        return bd.floatValue();
+    }
+
+    /**
+     * Take a group and a slot and return a percentage of
+     * the number of slot passed on the number of slot in the future
+     * for the group
+     * it is a rough estimate and doesn't take account the difference of length between slot
+     *
+     * @param group
+     * @param slot
+     * @return
+     */
+    public static float getPercentageOf(Group group, Slot slot) {
+        int total = 1;
+        int passed = 0;
+        List<Slot> listOfSlot;
+        SlotDAO dao = SlotDAO.getInstance();
+        try {
+            listOfSlot = dao.getSlotOfGroup(group);
```

Comment:
why using a group, shouldn't it be list of subject ?

Reply:
Why ? You want the rest of the slot for a subject for a group we might have several group with the same subject(physique in L1 for instance)

Reply:
We don't want to evaluate the number of hours for many groups, only for the one group the connected student is in.
So in this case list of groups is not relevant, using only one group is what we need

Reply:
but I don't use a listOfGroup but just the group of the student